### PR TITLE
Fix Lifecycle and Signal  docs

### PR DESCRIPTION
**Pull request by prudrabhat**
## Description

Fix Lifecycle and Signal  docs and fix links.

## Motivation and Context

Fix Lifecycle and Signal  docs and fix links.

## How Has This Been Tested?

Ran `yarn dev` locally to verify formatting and links

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/pages/documentation/mobile-core/lifecycle/tabs/index.md [hunk 1]
```diff
@@ -58,27 +66,18 @@ import 'package:flutter_acpcore/flutter_acplifecycle.dart';
 1. Register the Lifecycle extension:
 
 ```java
-public class TargetApp extends Application {
-
+public class MobiletApp extends Application {
```
Comment (addb) [inline src/pages/documentation/mobile-core/lifecycle/tabs/index.md:69]: typo: MobileApp*

File: src/pages/documentation/mobile-core/lifecycle/tabs/api-reference.md [hunk 2]
```diff
@@ -1,7 +1,15 @@
-<Variant platform="android" api="extension-version" repeat="2"/>
+<Variant platform="android" api="extension-version" repeat="5"/>
 
 #### Java
 
+**Syntax**
+
+```java
+public static String extensionVersion()
```
Comment (addb) [inline src/pages/documentation/mobile-core/lifecycle/tabs/api-reference.md:8]: @NonNull for return type?

Comment (prudrabhat) [reply src/pages/documentation/mobile-core/lifecycle/tabs/api-reference.md:8]: The current code should have had the NonNull annotation but looks [like it was a miss](https://github.com/adobe/aepsdk-core-android/blob/main/code/lifecycle/src/phone/java/com/adobe/marketing/mobile/Lifecycle.java#L30) . I left it out here to be true to what the code.

**Review by praveek — APPROVED** [review]

File: src/pages/documentation/mobile-core/lifecycle/tabs/index.md [hunk 3]
```diff
@@ -1,9 +1,17 @@
-<Variant platform="android" task="add" repeat="2"/>
+<Variant platform="android" task="add" repeat="4"/>
 
-Import the library:
+Add the Lifecycle extension and it's dependency, the [Mobile Core](../index.md) extension to your project using the app's Gradle file.
```
Comment (calebk1) [inline src/pages/documentation/mobile-core/lifecycle/tabs/index.md:3]: "... and its dependency..."

File: src/pages/documentation/mobile-core/lifecycle/tabs/index.md [hunk 4]
```diff
@@ -1,9 +1,17 @@
...
 
 ```java
-   import com.adobe.marketing.mobile.*;
+implementation 'com.adobe.marketing.mobile:core:2.+'
+implementation 'com.adobe.marketing.mobile:lifecycle:2.+'
+```
+
```
Comment (calebk1) [inline src/pages/documentation/mobile-core/lifecycle/tabs/index.md:9]: Are we also including the note about not using relative things here too?

Comment (prudrabhat) [reply src/pages/documentation/mobile-core/lifecycle/tabs/index.md:17]: I missed adding that! Thanks for catching this 😄 . Added it now.

**Review by calebk1 — APPROVED** [review]
LGTM!

**Review by addb — APPROVED** [review]

**Pull request merged**